tests/test_benchmark.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+from os import path
+from conftest import skipif
+from devito import configuration
+from subprocess import check_call
+
+
+pytestmark = skipif(['yask', 'ops'])
+
+
+@pytest.mark.parametrize('mode', ['bench'])
+@pytest.mark.parametrize('problem', ['acoustic', 'tti', 'elastic', 'viscoelastic'])
+def test_bench(mode, problem):
+    """
+    Test combinations of modes and problems.
+    Testing for both OpenMP enabled or disabled.
+    """
+    if configuration['openmp']:
+        nthreads = configuration['platform'].cores_physical
+    else:
+        nthreads = 1
+
+    command_bench = ['python', 'benchmarks/user/benchmark.py', mode, '-P', problem,
+                     '-d', '16', '16', '16', '--tn', '4', '-x', '1']
+    check_call(command_bench)
+
+    dir_name = 'results/'
+
+    base_filename = problem
+    filename_suffix = '.json'
+    arch = 'arch[unknown]'
+    shape = 'shape[16,16,16]'
+    nbl = 'nbl[10]'
+    tn = 'tn[4]'
+    so = 'so[2]'
+    to = 'to[2]'
+    dse = 'dse[advanced]'
+    dle = 'dle[advanced]'
+    at = 'at[aggressive]'
+    nt = path.join('nt[' + str(nthreads) + ']')
+    mpi = 'mpi[False]'
+    np = 'np[1]'
+    rank = 'rank[0]'
+    bkend = 'bkend[core]'
+
+    bench_corename = path.join('_'.join([base_filename, arch, shape, nbl, tn,
+                               so, to, dse, dle, at, nt, mpi, np, rank]))
+
+    bench_filename = path.join(dir_name + bench_corename + filename_suffix)
```

Comment:
this should be:

```
path.join(dir_name, bench_corename, filename_suffix)
```

Otherwise it's totally useless to use `path.join`

Reply:
ignore the comment above, I think? but why do you need path.join ?

Reply:
Well, bad leftovers...